eureka-client/src/main/java/com/netflix/discovery/converters/StringCache.java

```diff
@@ -0,0 +1,38 @@
+package com.netflix.discovery.converters;
+
+import java.util.Map;
+import java.util.WeakHashMap;
+
+/**
+* @author Tomasz Bak
+*/
+public class StringCache {
+
+    public static final int LENGTH_LIMIT = 38;
```

Comment:
StringCache implementation follows xstream's StringConverter which has this optimization. Processing long strings is expensive, and probability of having duplicates decreases with increased string length.
This PR was a quick prototype to check how big is the benefit of pooling strings. We should revise it further if we agree we want this change.